packages/salesforcedx-utils/src/cli/cliVersionStatus.ts

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright (c) 2023, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+import { execSync } from 'child_process';
+import * as semver from 'semver';
+
+export enum CliStatusEnum {
+  SFv2 = 1,
+  outdatedSFDXVersion = 2,
+  onlySFv1 = 3,
+  cliNotInstalled = 4,
+  bothSFDXAndSFInstalled = 5,
+  SFDXv7Valid = 6
+}
+
+export class CliVersionStatus {
+
+  public getSfdxCliVersion(): string {
```

Comment:
@daphne-sfdc this function and the next have the same logic, with diff in the command being run. See if this can be reduced to one function that manages the execSync and then use the current functions to call into the new function passing command to run.